dev-requirements.txt

```diff
@@ -1,5 +1,5 @@
-git+https://github.com/dbt-labs/dbt-adapters.git@main
-git+https://github.com/dbt-labs/dbt-adapters.git@main#subdirectory=dbt-tests-adapter
+git+https://github.com/dbt-labs/dbt-adapters.git@microbatch-behavior-flag
+git+https://github.com/dbt-labs/dbt-adapters.git@microbatch-behavior-flag#subdirectory=dbt-tests-adapter
```

Comment:
reminder: let's make sure we bump the lower bound pin of `dbt-adapters` in setup.py as part of this PR once the dbt-adapters branches are merged.